projects/generate_background/generate_background.py

```diff
@@ -0,0 +1,87 @@
+import logging
+from pathlib import Path
+from typing import List
+
+import h5py
+import numpy as np
+from gwdatafind import find_urls
+from gwpy.segments import DataQualityDict
+from gwpy.timeseries import TimeSeries
+from hermes.typeo import typeo
+
+
+@typeo
+def main(
+    start: float,
+    stop: float,
+    ifos: List[str],
+    sample_rate: float,
+    channel: str,
+    frame_type: str,
+    segment_name: str,
+    minimum_length: float,
+    outdir: Path,
+):
+    """Generates background data for training BBHnet
+
+    Args:
+        start: start gpstime
+        stop: stop gpstime
+        ifos: which ifos to query data for
+        outdir: where to store data
+    """
+
+    # query segments for each ifo
+    # I think a certificate is needed for this
+    segments = DataQualityDict.query_dqsegdb(
```

Comment:
I think you're right. This should probably be fine for now if we're just running this from LDG. Longer term if we want to run unit tests from a GitHub runner we it looks like the underlying call `gwpy` makes (through `dqsegdb`) to get credentials is [this](https://gwdatafind.readthedocs.io/en/stable/api/gwdatafind.utils.find_credential.html). So we can think about adding [Kerberos credentials](https://computing.docs.ligo.org/guide/auth/#ligo-kerberos) as a secret to the GitHub workflow then use it to generate an [X.509 credential](https://computing.docs.ligo.org/guide/auth/#ligo-x509) at a location saved to the appropriate environment variable. Not that we need any of this right now, just documenting it here for posterity.